File: backend/src/utils/apiResponse.ts
class ApiResponse<T> {
  statusCode: number;
  success: boolean;
  message: string;
  data: T;

  constructor(statusCode: number, success: boolean, message: string, data: T) {
    this.statusCode = statusCode;
    this.success = success;
    this.message = message;
    this.data = data;
  }
}

export { ApiResponse };

/* class ApiResponse<T> {
  constructor(
    public statusCode: number,
    public success: boolean,
    public message: string,
    public data: T
  ) {}
} */

File: java-backend/src/main/java/java_backend/utils/ApiResponse.java
package java_backend.utils;

import lombok.AllArgsConstructor;
import lombok.Data;

@Data
@AllArgsConstructor
public class ApiResponse {
    private boolean success;
    private int statusCode;
    private String message;
    private Object data;

    public ApiResponse() {
        // Default constructor for frameworks and serialization.
    }

    public ApiResponse(boolean success, int statusCode, String message, Object data) {
        this.success = success;
        this.statusCode = statusCode;
        this.message = message;
        this.data = data;
    }
}